applications/dashboard/controllers/api/MediaApiController.php

```diff
@@ -111,6 +111,14 @@ private function doUpload(UploadedFile $upload, $type) {
                 }
         }
 
+        // image dimensions are higher than limit, it needs resizing
+        if (c("ImageUpload.Limits.Enabled")) {
+            if ($media['ImageWidth'] > c("ImageUpload.Limits.Width") || $media['ImageHeight'] > c("ImageUpload.Limits.Height")) {
+                $imageResizer = new \Vanilla\ImageResizer();
```

Comment:
This should be dependency injected. The configuration should be injected as well.